donation_track_backend/api.py

```diff
@@ -104,14 +104,18 @@ def post(self):
     @marshal_with(resource_fields.donation_resource_fields)
     def get(self):
         args = request.args
-        try:
-            schemas.UserNameSchema().load(args["name"])
-        except ValidationError as err:
-            abort(message=err.messages, http_status_code=409)
-        donations = db_operations.get_donations_by_user(args["name"])
-        if donations is None:
-            abort(message="Donation not found", http_status_code=404)
-        return donations
+        if "donee_id" in args:
+            donations = db_operations.get_donations_by_donee(args["donee_id"])
+            if donations is None:
+                abort(message="Donation not found", http_status_code=404)
+            return donations
+
+        if "user_name" in args:
+            donations = db_operations.get_donations_by_user(args["user_name"])
+            if donations is None:
+                abort(message="Donation not found", http_status_code=404)
+            return donations
+        abort(message="donee_id or user_name arguments are missing", http_status_code=404)
```

Comment:
What if the client provides both donee_id and user_name?

Also I don't think the http status code `404` is appropriate in case `donee_id` and `user_name` are missing.